_tools/release_cycle_template.svg.jinja

```diff
@@ -42,40 +48,150 @@
         {% endif %}
     {% endfor %}
 
-    {% for version in versions %}
-        {% set y = loop.index * LINE_HEIGHT %}
+    <!-- Gradient mask to fade out end-of-life versions -->
+    <mask id="release-cycle-mask-{{ id_key }}">
+        <rect
+            x="0"
+            y="0"
+            width="{{ LEGEND_WIDTH * SCALE }}"
+            height="{{ diagram_height * SCALE }}"
+            fill="black"
+        />
+        <rect
+            x="{{ (LEGEND_WIDTH - RIGHT_MARGIN) * SCALE }}"
+            y="0"
+            width="{{ RIGHT_MARGIN * SCALE }}"
+            height="{{ diagram_height * SCALE }}"
+            fill="url(#release-cycle-mask-gradient-{{ id_key }})"
+        />
+        <rect
+            x="{{ (LEGEND_WIDTH ) * SCALE }}"
+            y="0"
+            width="{{ diagram_width * SCALE }}"
+            height="{{ diagram_height * SCALE }}"
+            fill="white"
+        />
+    </mask>
 
-        <!-- Legend on the left -->
-        <text
-            class="release-cycle-version-label"
-            x="{{ 0.5 * SCALE }}"
-            y="{{ y * SCALE }}"
-            font-size="{{ SCALE }}"
-        >
-            Python {{ version.key }}
-        </text>
+    {% for version in versions %}
+        {% set y = version.y * LINE_HEIGHT %}
 
         <!-- Colourful blob with a label -->
         {% set start_x = date_to_x(version.first_release_date) %}
         {% set end_x = date_to_x(version.end_of_life_date) %}
-        {% set mid_x = (start_x + end_x) / 2 %}
+
+        <!-- bugfix status needs a security tail.
+            Draw the rectangle with two path elements instead.
+            Thanks Claude.ai for the conversion.
+        -->
+        {% set half_x = [end_x, date_to_x(version.start_security_date)]|min %}
+        {% set height = 1.25 * SCALE %}
+        {% set left_width = (half_x - start_x) * SCALE %}
+        {% set right_width = (end_x - half_x) * SCALE %}
+        {% set left_x = start_x * SCALE %}
+        {% set middle_x = half_x * SCALE %}
+        {% set right_x = half_x * SCALE %}
+        {% set recty = (y - 1) * SCALE %}
```

Comment:
Little odd var name, maybe `rect_y` instead? Or what is this?
```suggestion
        {% set recty = (y - 1) * SCALE %}
```

Reply:
Changed.